### PR TITLE
python310Packages.canals: init at 0.2.2

### PR DESCRIPTION
**Pull request by happysalada**
###### Description of changes

package used in haystack 
ref https://github.com/deepset-ai/canals/blob/main/pyproject.toml

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/canals/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+  # Pypi source package doesn't contain tests
+  src = fetchFromGitHub {
+    owner = "deepset-ai";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-dF0bkY4DFJIovaseNiOLgF8lmha+njTTTzr2/4LzZEc=";
+  };
+
+  disabled = pythonOlder "3.7";
```
Comment (tjni) [inline pkgs/development/python-modules/canals/default.nix:34]: Nit: Think this should be `pythonOlder "3.8"`.

**Review by tjni — COMMENTED** [review]
Could we run the tests? Adding the following works on my aarch64-darwin machine:

```
  nativeCheckInputs = [
    pytestCheckHook
  ] ++ passthru.optional-dependencies.mermaid;
```
(also need to add `rec` if taking this approach)

**Comment by happysalada** [comment]
Thank you for coming around!
I fixed your comments. Do you mind not using `rec`. I personally don't like it because it doesn't make variable origin apparent.

**Comment by tjni** [comment]
> Do you mind not using `rec`. I personally don't like it because it doesn't make variable origin apparent.

I don't mind at all!

I didn't realize the tests need the Internet (I still need to turn on the sandbox on my aarch64-darwin machine somehow). Let me take a look.

**Comment by happysalada** [comment]
I have the same problem. The tests passed on my x86 darwin but i dont think the sandbox was enabled.

**Comment by tjni** [comment]
Tested adding the following seems to do the trick:

```
disabledTestPaths = [
  # requires internet connection to mermaid.ink
  "test/pipelines/integration"
];
```

**Comment by happysalada** [comment]
fixed. Thanks a lot for your active help on this, very appreciated, if I can ever help you on anything, happy to be pinged.

**Review by tjni — APPROVED** [review]

**Pull request merged**